mysite/w6/beanstalk.py

```diff
@@ -18,18 +21,19 @@ def __get_beanstalk_data(raw):
     else:
         return []
 
+
 def section_beanstalk():
     beanstalk_bought = "Gold Food Beanstalk" in session_data.account.jade_emporium_purchases
     upgrade_bought = "Supersized Gold Beanstacking" in session_data.account.jade_emporium_purchases
     beanstalk_data = __get_beanstalk_data(session_data.account.raw_data)
 
-    if not (beanstalk_bought or beanstalk_data):
+    if not (beanstalk_bought and beanstalk_data):
         return AdviceSection(
             name="Giant Beanstalk",
             tier="",
             header="Come back once you've bought the \"Gold Food Beanstalk\" from the Jade Emporium",
             picture="Jade_Vendor.gif",
-            collapse=False
+            collapse=False,
```

Comment:
collapse=True, otherwise we get an ugly empty spacing between sections (my bad initially)